Epsilon/Components/KpiMatrixComponent.cs

```diff
@@ -0,0 +1,192 @@
+using DocumentFormat.OpenXml;
+using DocumentFormat.OpenXml.Packaging;
+using DocumentFormat.OpenXml.Wordprocessing;
+using Epsilon.Abstractions;
+using Epsilon.Abstractions.Components;
+
+namespace Epsilon.Components;
+
+public class KpiMatrixComponent : AbstractCompetenceComponent
+{
+
+    public KpiMatrixComponent(IAsyncEnumerable<LearningDomainSubmission> submissions, IEnumerable<LearningDomain?> domains, IEnumerable<LearningDomainOutcome> outcomes)
+        : base(submissions, domains, outcomes)
+    {
+    }
+
+    public override async Task<Body> AddToWordDocument(MainDocumentPart mainDocumentPart)
+    {
+        var body = new Body();
+        // Create a table, with rows for the outcomes and columns for the assignments.
+        var table = new Table();
+
+
+        // Set table properties for formatting.
+        table.AppendChild(new TableProperties(
+            new TableWidth { Width = "0", Type = TableWidthUnitValues.Auto, }));
+
+        // Calculate the header row height based on the longest assignment name.
+        var headerRowHeight = 0;
+        var anyAssignments = false;
+
+        await foreach (var sub in Submissions)
+        {
+            anyAssignments = true;
+            headerRowHeight = Math.Max(headerRowHeight, sub.Assignment?.Length ?? 10);
+
+        }
+
+        if (anyAssignments)
+        {
+            headerRowHeight *= 111;
+        }
+
+        // Create the table header row.
+        var headerRow = new TableRow();
+        headerRow.AppendChild(new TableRowProperties(new TableRowHeight { Val = (UInt32Value)(uint)headerRowHeight, }));
+
+        // Empty top-left cell.
+        headerRow.AppendChild(CreateTableCellWithBorders("2500", new Paragraph(new Run(new Text("")))));
+        var index = 0;
+        await foreach (var sub in Submissions)
+        {
+            var cell = CreateTableCellWithBorders("100");
+
+            if (cell.FirstChild != null)
+            {
+                cell.FirstChild.Append(new TextDirection { Val = TextDirectionValues.TopToBottomLeftToRightRotated, });
+
+                cell.Append(new Paragraph(new Run(new Text(sub.Assignment ?? "Not found"))));
+                cell.FirstChild.Append(new Shading
```

Comment:
Shading is a tableCellProperty.  following the “child” [list](https://learn.microsoft.com/en-us/dotnet/api/documentformat.openxml.wordprocessing.tablecellproperties) of TableCellPropery. `shd` will need to be defined before `tcBorders` TableCellBorders